### ISSUE TITLE
Replicate the issue with invalid URL handling without Vaadin

### ISSUE THREAD

**Issue by denis-anisimov** [vaadin/spring#361]
This is the description of the problem:

https://github.com/vaadin/spring/issues/331#issuecomment-412429056

It happens if there is no hack with Dispatcher servlet mapped to the root.
If the mapping of VaadinServlet is the root then there is a forwarding controller which 
forwards all requests to the root to the VaadinServlet initially mapped to totally different URI.

With this forwarding the URL are rewritten in the incorrect way ( see https://github.com/vaadin/spring/issues/331#issuecomment-412429056) and VaadinServlet just doesn't work ( if Dispatcher servlet is not mapped to the root).

We should replicate the same thing without Vaadin.
Create a Spring ticket once this is done.

**Comment by denis-anisimov** [comment]
I've made a ticket for Spring:
https://jira.spring.io/browse/SPR-17457

But I think they won't do anything with it. Apparently this behavior is done intentionally for some reason.

Now I understand what happens with Spring and Vaadin in case of forwarding.

Shortly: we should do something similar to https://github.com/vaadin/spring/blob/2.1/vaadin-spring/src/main/java/com/vaadin/spring/server/SpringVaadinServletRequest.java

Here is what happens:
- the URL is "http://localhost:8080/some_path".
- Spring applies forwarding.
- With forwarding `HttpServletRequest::getServletPath` returns `/some_path` and `null` for `pathInfo` (without forwarding it would be : `HttpServletRequest::getServletPath` returns `""` and  `getPathInfo()`  returns `/some_path`).
- The `VaadinServlet::service`  method first calls `handleContextOrServletRootWithoutSlash`.
- The latter method checks exactly the situation when `pathInfo` returns `null` and `servletPath` doesn't end with slash.
- As a result the redirect is applied to the URL with added trailing slash: `http://localhost:8080/some_path/` . 
- The path becomes incorrect and at least our `WebJarServer` is not able to locate the web jars using this url.
- application is broken because it least has to load `webcomponents-loader.js`.

My hack with dispatcher servlet makes the `HttpServletRequest` working in the expected way. That's why everything works fine when dispatcher servlet is overriden.

So: 
- I made a ticket for Spring. We can see what they say about it. But most likely we will have to live with this.
- I suggest to make a workaround on out side similar to the FW8 with replacing HTTP request .
